### DISCUSSION TITLE
Idea: More than 8 apps by switching to circular app layout

### DISCUSSION THREAD

**Discussion by aaronmw** [rxhanson/Charmstone-Community#44] [Ideas]
Just occurred to me that the 8-app limit isn't really necessary if you switch to a circular layout. You could evenly space any number of app icons around a circle, maybe only if the number of apps exceeds 8? To me, I'd much prefer this to the near / far combination because I could simply remember which "time" the app was at on the entire circle. It'd also be more intuitive and visually pleasing for odd-numbered icon counts.

Anyway... Food for thought. Here's a picture of what I'm talking about:

![image](https://github.com/rxhanson/Charmstone-Community/assets/1236267/4dc7fd1e-9fba-4b0d-b8c3-4a48e40248de)

**Comment by rxhanson** [comment]
Thanks for the detailed feedback!

Charmstone is fundamentally built around this design of 8 apps surrounding it, with the regions being the trigger areas (not necessarily just the icons). This was an early design choice to facilitate faster app switching without having to land the cursor directly on the icon. Adding more trigger regions does not work very well when there is more than 8, meaning farther distance that your cursor would have to travel to distinguish regions, and the outcome of this would inevitably be sacrificing regions to go with this alternate design. I'm not opposed to this idea, but just trying to communicate that this would be a fundamental shift to how the app works (and therefore a nontrivial change to add in). I do really like the idea of the clock with 12 positions, since that's already an established mental model. 

I've often thought about a Charmstone v2 where there is some degree of per-app window switching included, and I might be inclined to build out the clock concept at that time. Unfortunately, I don't have a timeframe for this.

**Reply by aaronmw** [reply]
Yeah, that makes some sense. I thought about how I'd implement it myself and figured the math would be fairly simple. A matter of dividing 360 degrees into `n` regions (n = number of icons), then calculating the angle of the line from the origin (where the mouse was when the ring was drawn) and the cursor as it moves around to determine which "wedge" the line falls into.

There are a bunch of video games (first-person shooter types) that use "weapon wheels" and stuff to afford what is basically the same concept as this. I was playing FarCry the other day and when I tried Charmstone, I thought "hmm, I think this idea + FarCry would be sweeeet" :P 

Thanks for considering! Wish I had any idea how to build native apps like this. I'm limited to web apps / PWA's / anything I can build with React Native :(

**Comment by enieuwy** [comment]
I like this idea, but not because it can add more slots. Rather, my slight annoyance is that reaching the corner apps requires longer (or shorter) mouse travel. This is because, at a certain diagonal mouse travel, the selector can jump between the corner and the two adjacent sides. Often this results in selecting an adjacent app rather than the corner app.

Hope that explanation makes sense.

If all apps were equidistant from the centre (i.e., a circle), this problem would be resolved and I'd be delighted.

**Reply by rxhanson** [reply]
I appreciate the feedback, and still would like to add the circle in a future update (as always, time is the limiting factor). In terms of distance for reaching the apps, I think the gain is mostly psychological by switching to a circle. Charmstone is built using pie of a circle calculations for determining which item is selected, except for when the cursor is over an item. This is what's causing the jumping - it's just hard to see because there's no visual delineation between the slices. In my usage, I move fast and the cursor is almost always moved past the icon. The main benefit to having the square is that the window is smaller (and I actually quite like the aesthetic over the circle). I'm definitely not trying to downplay your opinion, but just looking to clarify that I have spent a lot of time thinking about this and there's reasoning as to why the app is the way that it is. I understand the desire, though, and would like to tackle this in a v2 of the app, along with a loooong list of other new ideas that I've accumulated over the years :)

**Comment by enieuwy** [comment]
Thanks for your response! That makes sense, and I can probably adapt my usage given that knowledge. Initially I thought the circle aesthetic would be better, but I think you're right given the icons are square.

________________________________
